core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/enclave_plan_capabilities/enclave_plan_capabilities_test.go

```diff
@@ -0,0 +1,68 @@
+package enclave_plan_capabilities
+
+import (
+	"encoding/json"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/service"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+const (
+	filesArtifactContentStr = "my-artifact-test-content"
+)
+
+func TestEnclavePlanCapabilitiesMarshallers(t *testing.T) {
+	originalEnclavePlanCapabilities := getEnclavePlanCapabilitiesForTest()
+
+	marshaledEnclavePlanCapabilities, err := json.Marshal(originalEnclavePlanCapabilities)
+	require.NoError(t, err)
+	require.NotNil(t, marshaledEnclavePlanCapabilities)
+
+	// Suppressing exhaustruct requirement because we want an object with zero values
+	// nolint: exhaustruct
+	newEnclavePlanCapabilities := &EnclavePlanCapabilities{}
+
+	err = json.Unmarshal(marshaledEnclavePlanCapabilities, newEnclavePlanCapabilities)
+	require.NoError(t, err)
+
+	require.Equal(t, originalEnclavePlanCapabilities, newEnclavePlanCapabilities)
+	require.Equal(t, filesArtifactContentStr, string(newEnclavePlanCapabilities.GetFilesArtifactMD5()))
```

Comment:
I'd personally prefer to have more simple and explicit test where you just manually write the JSON.
Those 2-ways test are nice but they are not `unit` per-se. So basically if both the marshaller and the unmarshaller forget about a field, the test would still pass but you would not persist this field